homeassistant/components/comelit/coordinator.py

```diff
@@ -138,6 +153,16 @@ def __init__(
         self.api = ComelitVedoApi(host, port, pin)
         super().__init__(hass, entry, VEDO, host)
 
-    async def _async_update_system_data(self) -> dict[str, Any]:
+    async def _async_update_system_data(
+        self,
+    ) -> dict[str, Mapping[int, ComelitVedoAreaObject | ComelitVedoZoneObject]]:
```

Comment:
Can we use a typed dict here so that we know if it's a dict of area objects or a dict of zone objects being accessed based on the top string key being used?

Reply:
Good idea, done!